### ISSUE TITLE
Clarify requirements when Mozilla rejects an audit

### ISSUE THREAD

**Issue by wthayer** [mozilla/pkipolicy#146]
In bug 1374381, a CA may not have fully understood Mozilla's expectations when an audit is rejected. We could make this more explicit by adding the following to section 3.1.3: Should Mozilla reject an audit statement for any reason, it is the CA's responsibility to supply a replacement audit statement covering the full period and scope required by this policy. Mozilla may require the use of a different Auditor.

**Comment by BenWilson-Mozilla** [comment]
I like this proposal - let's put something substantively similar at the end of [section 3.1.3](https://www.mozilla.org/en-US/about/governance/policies/security-group/certs/policy/#313-audit-parameters).

**Comment by dzacharo** [comment]
Let's assume an audit report covered 1 year, the CA submited the report covering the audit period of 1 year back, and Mozilla rejected the audit and/or the auditor.

Then the CA would be asked to find another auditor and perform a new audit covering the same audit period, but until this happens, time doesn't stop so we have now entered year 2, therefore the audit period covered by this new audit is longer than 1 year.

This proposal implies that Mozilla would accept a new audit that would cover more than 1 year back. Is that an accurate understanding?

Just to be clear, I fully support this and assuming that all audit records are kept by the CA, it should be perfectly doable for an auditor to cover evidence for longer audit periods (in rare circumstances).

**Comment by sleevi** [comment]
> Let's assume an audit report covered 1 year, the CA submited the report covering the audit period of 1 year back, and Mozilla rejected the audit and/or the auditor.
> 
> Then the CA would be asked to find another auditor and perform a new audit covering the same audit period, but until this happens, time doesn't stop so we have now entered year 2, therefore the audit period covered by this new audit is longer than 1 year.

No, this is not correct. The audit period would still be a year, so potentially multiple audit reports would be needed. The WebTrust auditors can report on past events, within limits due to professional standards. ETSI doesn’t have an equivalent concept, by very fundamental design of the ISO standards it’s based on, and that was an intentional decision by ETSI (well, tracing back to CEN as well).


> This proposal implies that Mozilla would accept a new audit that would cover more than 1 year back. Is that an accurate understanding?

No, I don’t believe it is. This is an active area of difference between WebTrust (being based within accounting standards) and ETSI (being based on manufacturing and ISMS)

**Comment by dzacharo** [comment]
Well, this is a technicality (having 1 audit report that covers more than 1 year, or 2 audit reports that have a max of 1 year each).

To the best of my knowledge, this "difference" you repeatedly call out, doesn't exist. Both schemes examine historical data and evidence to confirm compliance with applicable standards over a period of time **of the past**. 

I'm also unable to understand the point about the difference of "professional standards", as if WebTrust is the only scheme with such professional standards and ETSI/ISO/CEN auditors don't adhere to any such equivalent "professional/ethical/whatever" standards. Perhaps this is a big misunderstanding because I can't imagine you claiming that ETSI auditors do not have professional standards or that they are not being professional at what they do.

**Comment by sleevi** [comment]
> Well, this is a technicality (having 1 audit report that covers more than 1 year, or 2 audit reports that have a max of 1 year each).

No, it's not. It's a substantial difference that we haven't seemed to be able to make headway in reaching a shared understanding.

> To the best of my knowledge, this "difference" you repeatedly call out, doesn't exist. 

ETSI and CEN intentionally called out this difference when developing the ETSI ESI specifications. If I need to dig through the archives to provide you a citation for this, I can; I believe I previously provided links in the CA/B Forum.

> I'm also unable to understand the point about the difference of "professional standards", as if WebTrust is the only scheme with such professional standards and ETSI/ISO/CEN auditors don't adhere to any such equivalent "professional/ethical/whatever" standards. 

It actually is that simple. There aren't. 

There's not an equivalent framework with respect to reporting, by design, in the approach used by ETSI auditors. Here, equivalent means "to the same specificity and with the same goals". AICPA, CPA Canada, and more broadly, IFAC, have developed a set of specific standards with respect to specific reporting and how they're conducted.

This is very different than the model that ISO has developed, as adopted by EA and ETSI, which reflects the different goals. The model for ISO 17065-based frameworks is specifically to leave such "professional standards" specifications in the context of the relevant audit scheme derived from 17065-and-friends. This is reflective of the fact that it's attempting to be a generic framework for multiple industries, while the AICPA/CPA Canada/IFAC approaches are specifically tailored for certain environments and customer needs. With respect to the EA/ETSI approach, these standards are largely left to individual CABs, to evaluate with their NABs, within the framework of EA recognition. However, this is very different than the WebTrust approach, which obligates a centralized set of standards. ENISA itself has long recognized this as a concern with the ETSI ESI approach, and the inconsistencies between CABs, NABs, and the overall framework.

I suspect this is not the appropriate issue to discuss the history and approaches of WebTrust and ETSI, which requires understanding the history around the development and implementation of the eSignature directive. It's not unreasonable to expect that CAs are not familiar with this, as they are not the ones consuming these audits and having to keep track of. However, we can probably continue this discussion elsewhere, and I can provide further references for you to understand why they're fundamentally different.

**Comment by sleevi** [comment]
If you do need references, the footnotes in https://cabforum.org/wp-content/uploads/CABF45-Sleevi-Whats-Wrong-With-the-Ecosystem.pdf have many of the primary sources you can learn from, which try to trace through to the beginning for both WebTrust and ETSI.

**Comment by dzacharo** [comment]
I remember that presentation. I do recall that several participants, including Auditors, did not support some of the conclusions. I am not qualified and cannot analyze the core of this particular presentation. This is something only Auditors with enough experience, and having done a similar investigation, can perform.

There is one fact that is interesting to share here, and that is ISO 21188. This has been used as the basis of WebTrust for CAs. It seems like there is something common between the two audit schemes.

On the other hand, as you already know, the WebTrust Task Force mainly focuses on the WebTrust audit scheme while ETSI also develops technical standards, AND the corresponding audit schemes. Whenever there is a change in ISO 21188, the WebTrust Task Force analyzes these changes and decides what to incorporate in the WebTrust for CAs document, but to the best of my knowledge they do not propose or change the technical standards decided by ISO or the CA/B Forum.

I assume that representatives of the WebTrust Task Force, CPA Canada, ETSI and ACAB-c would be able to share more into this interesting discussion, as it seems to directly or indirectly affect Mozilla Policy. Whether this discussion should take place (or not), here or in m.d.s.p. is probably something to be decided by the Mozilla Module Owners.

**Comment by sleevi** [comment]
> I do recall that several participants, including Auditors, did not support some of the conclusions. 

It's unfortunate you walked away with that conclusion, considering that both WebTrust and ETSI auditors represented it was an accurate statement about their history, and their differences, even if there were minor quibbles. It feels an unfortunate intellectual trick to attempt to dismiss the whole over a minor quibble, and even more unfortunate to ignore the primary sources referenced, which was the purpose of mentioning it.

> I am not qualified and cannot analyze the core of this particular presentation. This is something only Auditors with enough experience, and having done a similar investigation, can perform.

This is, fundamentally, the problem, as it's not the case that this divination is left to Auditors. You can and should be qualified, particularly if you're going to engage in the compare/contrast, precisely because Audits are meant to be used by the Relying Party (i.e. Browsers), and so we have to know this. If you don't feel you understand enough, then you can certainly take the steps to read the primary sources, and begin your own journey of self-education into the nuances of differences. This is something **everyone** considering _consuming_ audits has to do at some point.

A Root Program is expected to be intimately familiar with these details, if they choose to accept audits from different schemes, precisely because the audits they accept are in furtherance of their goals. That is, there's no sense in using an audit scheme that is not useful to your goals, and no sense in accepting an audit scheme that has weaker guarantees than what you currently accept.

I, as an individual, am perfectly capable of coming up with my own audit scheme and licensing my own auditors, have my own auditing criteria, and that very well can provide more assurance than either WebTrust or ETSI. You might find that unfathomable, but for a number of years, that's precisely how the Mozilla Root Program operated, and it **did** provide greater assurances and transparency. This isn't dark arts, it's just a topic that requires appropriate background reading, research, and willingness to learn and ask questions.

> There is one fact that is interesting to share here, and that is ISO 21188. This has been used as the basis of WebTrust for CAs. It seems like there is something common between the two audit schemes.

That seems like a gross mischaracterization, and it's unclear if that was intentional or just a misunderstanding. Slides 39-42 certainly reference ISO 21188, but only in a dotted line fashion meant to denote "The authors were aware of this, rejected it, but still kept in mind," as the footnotes discuss. But that's like arguing that a Sherman tank is a perfectly acceptable weekend getaway vehicle, since they both have automobiles in common, despite being intended for very different purposes and having very different constraints.

>  while ETSI also develops technical standards, AND the corresponding audit schemes.

ETSI cannot be said to develop audit schemes. It develops technical standards to support an audit scheme, certainly, but the holistic realization of an audit scheme is left to individual CABs (for unqualified trust services) or the Supervisory Bodies (qualified trust services). Particularly for unqualified trust services, the supervision scheme established by the eIDAS Regulation has no bearing or justification to be considered part of the scheme.

> I assume that representatives of the WebTrust Task Force, CPA Canada, ETSI and ACAB-c would be able to share more into this interesting discussion, as it seems to directly or indirectly affect Mozilla Policy. Whether this discussion should take place (or not), here or in m.d.s.p. is probably something to be decided by the Mozilla Module Owners.

I'm sure there is room, but I think it's worth highlighting that the interesting discussion is more or less a result of continuing to insist certain elements about ETSI and ACAB-c that are factually false, while I patiently try to explain and provide primary references to demonstrate why they're false, and then you either disagreeing or shrugging and going "Who can really say for sure?". I would be thrilled if there were more engaged participation from folks who studied ETSI and WebTrust standards and history, and could engage in a discussion about the merits and aligning terminology.

However, all of this started because you stated something [egregiously wrong in interpretation](https://github.com/mozilla/pkipolicy/issues/146#issuecomment-686262344), I tried to correct it, and you [doubled down](https://github.com/mozilla/pkipolicy/issues/146#issuecomment-686265477) on misunderstanding what an audit period and report are, relative to WebTrust and ETSI. I suspect that this issue is best kept focused on the "simple" problem at hand, of requiring a new assessment if an auditor is disqualified, and that we otherwise ignore the digression about confusion about the audit period, and then audits themselves, and simply acknowledge your first comment was misunderstanding.

It is useful, however, to know that there was a misunderstanding, because it again highlights that the approach by ETSI of developing a certification scheme yet again means that there are complications for ensuring browser security needs are met, and the rejection of audits from an ETSI auditor will likely involve a careful walkthrough and discussion with their **new** auditor about what is expected in order to prevent the new report from being rejected. But that's a bridge we can cross, because it's unlikely that we can nail down "Here's how a CAB should do its job" briefly in a policy.

If you'd like to again litigate a misunderstanding of the audit criteria, I'm more than happy to continue to explain to you on m.d.s.p., but my only hope is that you'll carefully examine the primary sources I provided you. This was no small feat, requiring weeks of research and significant cross-referencing of applicable standards, and I truly hope you'll engage with the same rigor. However, regardless of your commitment to better understanding WebTrust and ETSI, I do hope you carefully review some of the sources listed on Slides 51 - 54, and with a special emphasis on the [ENISA report on audit standards](https://www.enisa.europa.eu/publications/tsp_standards_2015/at_download/fullReport) that includes the EA/ETSI disposition on comments on Page 15 / Section 2.1.2. Fundamentally, the view of CABs as, well, certification bodies, is directly and explicitly at odds with the WebTrust approach, and this fundamentally alters the design, maintenance, and management. Once you've had a chance to digest and internalize this information, I'd be happy to continue further on m.d.s.p.

**Comment by BenWilson-Mozilla** [comment]
I thought we might be able to address this issue at the same time we address qualified audits (Issue #155), but I realize now that we're talking about two different things.
